src/visitors/visitor_utils.cpp

```diff
@@ -51,12 +51,13 @@ std::string suffix_random_string(const std::set<std::string>& vars,
 std::string get_new_name(const std::string& name,
                          const std::string& suffix,
                          std::map<std::string, int>& variables) {
-    int counter = 0;
-    if (variables.find(name) != variables.end()) {
-        counter = variables[name];
-    }
-    variables[name] = counter + 1;
-    return (name + "_" + suffix + "_" + std::to_string(counter));
+    auto it = variables.insert({name, 0});
```

Comment:
didn't remember insert with the same key is a no-op!  :ok_hand:

Reply:
`std::map::emplace` would be even better! I will change that.

Reply:
Done in 66da7cc

Reply:
waaait a minute! I had a very long discussion with @ohm314 about the fact that in this case it is equivalent and he really prefers insert. I will use your opinion against him!

Reply:
I prefer `std::map::emplace` because in the case the insertion doesn't occur because the key is already there, the `std::pair<const Key, T>` constructed when calling `std::map::insert` has been built for nothing.

Reply:
My interpretation of Omar's comment was that when perf is not an issue (in trivial cases), one can use "natural" way of doing things. (now, where `emplace` or `insert` is natural to someone is a different question altogether :grin:)

Reply:
> one can use "natural" way of doing things

The _"natural" way of doing things_ evolves along with the C++ standard. There are good practices and should not be subject to interpretation. One should not prefer `insert` over `emplace` because the name is shorter or more indicative.

`std::map::emplace` is the new _natural way_ to perform this operation since C++11.
and IMO whether performance is critical or not should be out of the equation.

Because you never know how your code will be used or modified in the future, one should do its best to avoid the worst. For instance, someone could change the code to use a much bigger type for the key or value and someone else could decide later to call your code one zillions times, and suddenly you have a problem that would have been avoided if you followed the actual _"natural" way of doing things` right at the beginning.

Don't wait for problems to arise to start doing things the right way, especially when it is so easy to do so.

Reply:
Obviously what @tristan0x said above is right and I am not responding here to counter above comment but to clarify the context to my previous comment. Without context it's easy to miss-interpret.


> One should not prefer insert over emplace because the name is shorter or more indicative.

By natural I didn't meant _what I feel natual_ . In the past month when this question came up and I read through various posts to understand why & when to use emplace. Here is the [previous PR](https://github.com/BlueBrain/nmodl/pull/316/files/a6b95081e37f30b9802b0d7d9a30bc889d971a99#diff-eca115555591ff57c283de7d67303507) (click on show conversation) where this discussion started .

After reading this tip of the week from https://abseil.io/tips/112 : 

> For more expensive types, this may be a reason to use emplace_back() instead of push_back(), despite the readability and safety costs, but then again it may not. Very often the performance difference just won’t matter.
>
>As always, the rule of thumb is that you should avoid “optimizations” that make the code less safe or less clear, unless the performance benefit is big enough to show up in your application benchmarks.
>
>So in general, if both push_back() and emplace_back() would work with the same arguments, you should prefer push_back(), and likewise for insert() vs. emplace().

And that's when I thought it is not necessary to enforce such changes in very simple scenarios.